lib/amqp_client.js

```diff
@@ -378,11 +379,23 @@ AMQPClient.prototype.createReceiver = function(address, options, cb) {
   var linkPolicy = u.deepMerge({
     options: {
       name: linkName,
-      source: { address: address },
+      source: { address: address.name },
       target: { address: 'localhost' }
     }
   }, self.policy.receiverLink);
 
+  // if a subject has been provided then automatically set up a filter to match
+  // on that subject.
+  if (!!address.subject) {
+    var filterSymbol = (address.subject.indexOf('*') || address.subject.indexOf('#')) ?
+      'apache.org:legacy-amqp-topic-binding:string' :
```

Comment:
I think I'll move them into an exported module, since I'd like to be able to reuse these values for the short-hand filter format we discussed earlier today. Perhaps `require('amqp10').Filters` ?